### DISCUSSION TITLE
How to update source XML component via API?

### DISCUSSION THREAD

**Discussion by justscribe** [WeblateOrg/weblate#11178] [Q&A] [answered]
Hi Community,
I'm trying to update existing source language with new data by API.
Tried [PATCH](https://docs.weblate.org/en/latest/api.html#patch--api-components-(string-project)-(string-component)-) and [PUT](https://docs.weblate.org/en/latest/api.html#put--api-components-(string-project)-(string-component)-), but seems that is changing only the component settings, not the component content.
Component creation is done with [POST](https://docs.weblate.org/en/latest/api.html#post--api-projects-(string-project)-components-).
The file itself is a flat XML.
Maybe I've used it incorrectly, please help.
Thank you.

**Comment by nijel** [comment, marked as answer]
You can either [update individual units](https://docs.weblate.org/en/latest/api.html#patch--api-units-(int-id)-) or [upload whole file](https://docs.weblate.org/en/latest/api.html#post--api-translations-(string-project)-(string-component)-(string-language)-file-).

**Reply by justscribe** [reply]
Thank you.
